mmedit/models/base_models/base_conditional_gan.py

```diff
@@ -183,48 +183,70 @@ def forward(self,
 
         labels = self.data_sample_to_label(data_samples)
         if labels is None:
-            # num_batches = get_valid_num_batches(inputs)
             labels = self.label_fn(num_batches=num_batches)
 
         sample_model = self._get_valid_model(inputs)
-        if sample_model in ['ema', 'ema/orig']:
-            generator = self.generator_ema
-        else:  # sample model is `orig`
-            generator = self.generator
-        outputs = generator(noise, label=labels, return_noise=False)
-        outputs = self.data_preprocessor.destruct(outputs, data_samples)
-
-        if sample_model == 'ema/orig':
-            generator = self.generator
-            outputs_orig = generator(noise, label=labels, return_noise=False)
+        batch_sample_list = []
+        if sample_model in ['ema', 'orig']:
+            if sample_model == 'ema':
+                generator = self.generator_ema
+            else:
+                generator = self.generator
+            outputs = generator(noise, label=labels, return_noise=False)
+            outputs = self.data_preprocessor.destruct(outputs, data_samples)
+
+            # save to data sample
+            for idx in range(num_batches):
```

Comment:
Now I think we don't need this loop, we can add gen_sample(type: tensor) into datasample directly and then use datasample.split() to return the list of datasamples.